packages/ibm-products/src/components/SteppedAnimatedMedia/SteppedAnimatedMedia.js

```diff
@@ -0,0 +1,153 @@
+// Default values should be provided when the component needs to make a choice
+// or assumption when a prop is not supplied.
+
+// Default values for props
+const defaults = {
+  playStep: 0,
+};
+
+/**
+ * DO NOT USE. This component is for the exclusive use
+ * of other Novice to Pro components.
+ */
+export let SteppedAnimatedMedia = React.forwardRef(
+  (
+    {
+      // The component props, in alphabetical order (for consistency).
+
+      className,
+      /* TODO: add other props for SteppedAnimatedMedia, with default values if needed */
+      playStep = defaults.playStep,
+      filePaths,
+      // Collect any other property values passed in.
+      ...rest
+    },
+    ref
+  ) => {
+    const [jsonData, setJsonData] = useState([]);
+    const animRef = useRef();
+    const backupRef = useRef();
+    const localRef = ref ?? backupRef;
+    // load animation source
+    useEffect(() => {
+      const isJsonFile = (filePath) => filePath.includes('.json');
+      async function loadArtifact() {
+        if (filePaths && filePaths.every(isJsonFile)) {
+          const allJsonArtifacts = filePaths.map((el) => fetch(el));
+          const rawResponse = await Promise.all(allJsonArtifacts);
+          const jsonResponse = rawResponse.map((el) => el.json());
+          const fileData = await Promise.all(jsonResponse);
+          setJsonData(fileData);
+        }
+      }
+      loadArtifact();
+    }, [filePaths]);
+
+    useEffect(() => {
+      const prefersReducedMotion = window?.matchMedia
+        ? window.matchMedia('(prefers-reduced-motion: reduce)').matches
+        : true;
+      if (localRef.current) {
+        animRef.current?.destroy();
+        animRef.current = lottie.loadAnimation({
+          container: localRef.current,
+          renderer: 'svg',
+          animationData: jsonData[clamp(playStep, 0, jsonData.length - 1)],
+          loop: false,
+          autoplay: false,
+          rendererSettings: {
+            preserveAspectRatio: 'xMidYMid slice',
+          },
+        });
+        prefersReducedMotion
+          ? animRef.current?.goToAndStop(0)
+          : animRef.current?.goToAndPlay(0);
+      }
+
+      return () => animRef.current?.destroy();
+    }, [jsonData, localRef, playStep]);
+
+    if (!jsonData) {
+      return null;
+    }
+
+    return (
+      <div
+        {
+          // Pass through any other property values as HTML attributes.
+          ...rest
+        }
+        className={cx(
+          blockClass, // Apply the block class to the main HTML element
+          className, // Apply any supplied class names to the main HTML element.
+          // example: `${blockClass}__template-string-class-${kind}-n-${size}`,
+          {
+            // switched classes dependant on props or state
+            // example: [`${blockClass}__here-if-small`]: size === 'sm',
+          }
+        )}
+        ref={localRef}
+        role="main"
```

Comment:
Feel like we should be cautious about using `role="main"`[^1]. Was this for getting past the tests? If so, I’d recommend wrapping `SteppedAnimatedMedia` with a `<main>` tag for just the test since these are components and the landmark tests are a bit tricky otherwise.

[^1]: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/main_role

Reply:
Ah no, I believe this is just how the components are scaffolded when they are generated and I left it as is. This might make more sense as a role="img" or "presentation"?